backend/apps/ecommerce/models.py

```diff
@@ -19,6 +19,8 @@ class Product(models.Model):
     total_quantity = models.PositiveIntegerField()
     current_quantity = models.PositiveIntegerField(null=True)  # Set to total_quantity upon initialization
     max_buyable_quantity = models.PositiveIntegerField(default=1)
+    types = models.JSONField(default=list, blank=True)
+    sizes = models.JSONField(default=list, blank=True)
```

Comment:
I'll just preface this by saying I know this is a work in progress, and that there's nothing inherently wrong with having these as `JSONField`. _However_, it has an important and regrettable side effect resulting from a lot of (unfortunate) magic behind the scenes with `graphene-django`. `graphene-django` automatically maps Django model fields (in this case, `JSONField`) to appropriate GraphQL types, e.g. `models.PositiveIntegerField` is mapped to `Int` in GraphQL.

However, GraphQL does not have a `JSON`-type, and as a result, `graphene-django` adds a custom JSON [scalar](https://graphql.org/learn/schema/#scalar-types) (I highly recommend that you have a look at the linked docs, they're great). A JSON scalar has a major issue, which graphql-django summarizes well in the generated code
> Allows use of a JSON String for input / output from the GraphQL schema.
> Use of this type is *not recommended* as you lose the benefits of having a defined, static
> schema (one of the key benefits of GraphQL).

In short, we want to avoid this JSON scalar. To work around this, you have (at least) two options:
1. Define the field as an [`ArrayField`](https://docs.djangoproject.com/en/3.2/ref/contrib/postgres/fields/#arrayfield) which is a Postgres-specific field, see the linked docs. This is perfectly safe as we use a postgres database.
2. Keep it as a JSONField, but instead of adding `types` and `sizes` to `fields` in the `ProductType` in `backend/apps/ecommerce/types.py`, add two custom resolvers for types and sizes where you return a list of strings, rather than the poorly defined JSON.

I hope this comment is not too overwhelming, and it's of course (as with all things in this project), nothing too dangerous, but I think there's quite a bit of insight to gain here on the value proposition of GraphQL and how the different parts of the project interact. Just send me a DM on Slack if there are any questions 😊